### PR TITLE
refactor(redux-saga): Remove create slice saga

### PR DESCRIPTION
**Pull request by xrutayisire**
## Context

- Completes DT-1443 - Convert createSliceSaga to a simple TypeScript function

## The Solution

- Remove createSliceSaga
- Add the New Slice and Update Slices button in the header all the time (DESIGN)
- Create a "createSlice" simple TS function that don't know anything about state management
- Handle the custom logic in CreateSliceModal
- Update Redux store in CreateSliceModal

## Impact / Dependencies

- Another issue has been created to handle the removal of the getState call in the CreateSliceModal: DT-1453

## Checklist before requesting a review

- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [x] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/components/Forms/CreateSliceModal/CreateSliceModal.tsx [hunk 1]
```diff
@@ -1,62 +1,124 @@
...
+          };
+          await managerClient.customTypes.updateCustomType({
+            model: CustomTypes.fromSM(newCustomType),
+          });
+          saveCustomTypeSuccess(CustomTypes.fromSM(newCustomType));
+        }
+
+        // Redirect to the slice page
+        const variationId = newSlice.variations[0].id;
+        const sliceLocation = `/${libraryName.replace(
```
Comment (hypervillain) [inline packages/slice-machine/components/Forms/CreateSliceModal/CreateSliceModal.tsx]: I would prefer this to be an external function, a bit like you did for the CT tables

File: packages/slice-machine/src/features/slices/actions/createSlice.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    id: snakelize(sliceName),
+    type: "SharedSlice",
+    name: sliceName,
+    description: sliceName,
+    variations: [
+      {
+        id: DEFAULT_VARIATION_ID,
+        name: pascalize(DEFAULT_VARIATION_ID),
+        // TODO: What should this be?
+        docURL: "...",
```
Comment (hypervillain) [inline packages/slice-machine/src/features/slices/actions/createSlice.ts:57]: this property is unused

File: packages/slice-machine/src/features/slices/actions/createSlice.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        // TODO: What should this be?
+        version: "initial",
```
Comment (hypervillain) [inline packages/slice-machine/src/features/slices/actions/createSlice.ts:59]: this is set by the CT API, any value here is fine

File: packages/slice-machine/src/features/slices/actions/createSlice.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        description: pascalize(DEFAULT_VARIATION_ID),
+        // TODO: What should this be? The type requires it.
+        imageUrl: "",
```
Comment (hypervillain) [inline packages/slice-machine/src/features/slices/actions/createSlice.ts:64]: This is required in the model but stripped by SM to be compared to the remote slice. Empty string is fine I believe, we just need to make sure it's not saved in the JSON model

Comment (hypervillain) [reply packages/slice-machine/src/features/slices/actions/createSlice.ts:64]: (although saving it to the model should not cause any issue)

**Review by hypervillain — CHANGES_REQUESTED** [review]
Very cool

File: packages/slice-machine/components/Forms/CreateSliceModal/CreateSliceModal.tsx (hunk 1 above)
Comment (xrutayisire) [reply packages/slice-machine/components/Forms/CreateSliceModal/CreateSliceModal.tsx]: Done

File: packages/slice-machine/src/features/slices/actions/createSlice.ts (hunk 2 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/slices/actions/createSlice.ts:57]: Done

File: packages/slice-machine/src/features/slices/actions/createSlice.ts (hunk 3 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/slices/actions/createSlice.ts:59]: Done

File: packages/slice-machine/src/features/slices/actions/createSlice.ts (hunk 4 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/slices/actions/createSlice.ts:64]: Done

**Pull request merged**